File: output/CodeLlama/codenet/C++/Python/s023282516.py
# -*- coding: utf-8 -*-

def main():
    s = input()

    # 後ろから解くかわりにすべての文字列を「左右反転」する
    s = s[::-1]
    divide = ["dream", "dreamer", "erase", "eraser"]
    divide = [d[::-1] for d in divide]

    # 端から切っていく
    can = True
    for i in range(len(s)):
        can2 = False # 4 個の文字列たちどれかで divide できるか
        for d in divide:
            if s[i:i + len(d)] == d: # d で divide できるか
                can2 = True
                i += len(d) # divide できたら i を進める
        if not can2: # divide できなかったら
            can = False
            break

    if can: print("YES")
    else: print("NO")

if __name__ == "__main__":
    main()

File: output/CodeLlama/codenet/C++/Java_repair/s023282516.java
import java.util.Scanner;

public class s023282516 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String S = sc.next();

        // 後ろから解くかわりにすべての文字列を「左右反転」する
        S = new StringBuilder(S).reverse().toString();
        String[] divide = {"dream", "dreamer", "erase", "eraser"};
        for (int i = 0; i < divide.length; ++i) divide[i] = new StringBuilder(divide[i]).reverse().toString();

        // 端から切っていく
        boolean can = true;
        for (int i = 0; i < S.length();) {
            boolean can2 = false; // 4 個の文字列たちどれかで divide できるか
            for (int j = 0; j < divide.length; ++j) {
                String d = divide[j];
                if (S.substring(i, i + d.length()).equals(d)) { // d で divide できるか
                    can2 = true;
                    i += d.length(); // divide できたら i を進める
                }
            }
            if (!can2) { // divide できなかったら
                can = false;
                break;
            }
        }

        if (can) System.out.println("YES");
        else System.out.println("NO");
    }
}